lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_extent.rb

```diff
@@ -21,8 +23,15 @@ def self.unpack(xExtent, hResponseObj)
                   # :geographicExtents (optional)
                   # <xs:element name="geographicElement" type="gmd:EX_GeographicExtent_PropertyType" minOccurs="0" maxOccurs="unbounded"/>
                   xGeoElems = xExExtent.xpath(@@geoElemXPath)
-                  unless xGeoElems.nil?
-                     hExtent[:geographicExtents] = xGeoElems.map { |g| GeographicExtent.unpack(g, hResponseObj) }
+                  unless xGeoElems.empty?
+                     hExtent[:geographicExtents] = xGeoElems.map { |g| GeographicExtent.unpack(g, hResponseObj) }.compact
```

Comment:
do we want to compact the nil values?

Reply:
I think we should use it in situations where `self.unpack` reasonably returns `nil` compared to returning whatever hash the function is responsible for populating as a way to keep the containing array clean. i don't see any value in handling something like `[ nil, nil, nil ]`. this is something I was going to go back and do among some other things.

Reply:
returning nil instead of a hash also avoids unnecessary bloat of conditional statements figuring out if the thing is "empty"